src/lib/components/TextBox/Hooks.ts

```diff
@@ -1,10 +1,16 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { TextBoxProps } from "./DTOs";
 
 const useTextBox = (props: TextBoxProps) => {
   const { regex = "", length = 0, value = "", onChange } = props;
   const [input, setInput] = useState<string>(value);
 
+  useEffect(() => {
+    if (onChange) {
+      onChange(input);
+    }
+  }, [input]);
+
   const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
we do not need html element here, we just need the value so rewrite it into `(inputValue: string)` and remove the 15th line